stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java

```diff
@@ -0,0 +1,177 @@
+                name = entry.getName();
+                if (!folderToUnzip.isEmpty() && name.startsWith(folderToUnzip)) {
+                    int index = name.lastIndexOf('/') + 1;
+                    name = name.substring(index);
+                }
+                    if (entry.isDirectory()) {
+                        mkdirs(outdir, name);
+                        continue;
+                    }
+                    dir = getDirectoryPath(name);
+                    if (dir != null) {
+                        mkdirs(outdir, dir);
+                    }
+                    extractFile(zin, outdir, name);
+            }
+        } catch (IOException e) {
+            throw new BLangRuntimeException("I/O Exception when processing files");
+        } finally {
+            try {
+                if (zin != null) {
+                    zin.close();
+                }
+            } catch (IOException e) {
+                throw new BLangRuntimeException("I/O Exception when closing the input stream " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Extract files from the zipInputStream.
+     *
+     * @param in     zipInputStream object
+     * @param outdir output directory file
+     * @param name   name of the file
+     */
+    private static void extractFile(ZipInputStream in, Path outdir, String name) {
+        byte[] buffer = new byte[4096];
+        BufferedOutputStream out = null;
+        try {
+            Path resourcePath = Paths.get(outdir.toString()).resolve(name);
+            out = new BufferedOutputStream(new FileOutputStream(resourcePath.toString()));
+            int count;
+            while ((count = in.read(buffer)) != -1) {
+                out.write(buffer, 0, count);
+            }
+        } catch (FileNotFoundException e) {
+            throw new BLangRuntimeException("File not found to process " + outdir);
+        } catch (IOException e) {
+            throw new BLangRuntimeException("I/O Exception when closing the input stream " + outdir);
+        } finally {
+            if (out != null) {
+                try {
+                    out.close();
+                } catch (IOException e) {
+                    throw new BLangRuntimeException("I/O Exception when closing the input stream " + e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Create the directory name.
+     *
+     * @param name name of the directory
+     * @return directory name
+     */
+    private static String getDirectoryPath(String name) {
+        if (name != null) {
+            int s = name.lastIndexOf(File.separatorChar);
+            return s == -1 ? null : name.substring(0, s);
+        }
+        return null;
+    }
+
+    /**
+     * Make directories if they doesn't exists.
+     *
+     * @param outdir destination file
+     * @param path   path of the destination directory
+     */
+    private static void mkdirs(Path outdir, String path) {
```

Comment:
can you reuse https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createDirectories-java.nio.file.Path-java.nio.file.attribute.FileAttribute...-

Reply:
Reused Files.createDirectories with bc105af